arch/arm/boot/dts/imx6ul-ts7180.dts

```diff
@@ -0,0 +1,1020 @@
+};
+
+&uart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart2>;
+	status = "okay";
+};
+
+&uart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3>;
+	status = "okay";
+	uart-has-rtscts;
+};
+
+&uart4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart4>;
+	status = "okay";
+	uart-has-rtscts;
+	rts-gpios = <&gpio1 19 GPIO_ACTIVE_HIGH>; /* EN_485 is the TX enable signal */
+	linux,rs485-enabled-at-boot-time;
+};
+
+&uart5 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart5>;
+	status = "okay";
+};
+
+
+&uart6 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart6>;
+	status = "okay";
+};
+
+&uart7 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart7>;
+	status = "okay";
+};
+
+&uart8 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart8>;
+	status = "okay";
+};
+
+&usdhc1 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc1>;
+	pinctrl-1 = <&pinctrl_usdhc1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc1_200mhz>;
+	max-frequency = <208000000>;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	disable-wp;
+	cap-sd-highspeed;
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
+	sd-uhs-ddr50;
+	broken-cd = <1>;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&usdhc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc2>;
+	no-1-8-v;
+	disable-wp;
+	broken-cd = <1>;
+	enable-sdio-wakeup;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&i2c3 {
+	clock-frequency = <400000>;
+	pinctrl-names = "default", "gpio";
+	pinctrl-0 = <&pinctrl_i2c3>;
+	pinctrl-1 = <&pinctrl_i2c3_gpio>;
+	status = "okay";
+	scl-gpios = <&gpio3 6 GPIO_ACTIVE_HIGH>;
+	sda-gpios = <&gpio3 5 GPIO_ACTIVE_HIGH>;
+
+	gpio6: gpio@28 {
+		compatible = "technologic,ts4900-gpio";
+		reg = <0x28>;
+		#gpio-cells = <2>;
+		gpio-controller;
+		base = <160>;
+		ngpios = <64>;
+
+		gpio-line-names =
+			/* Outputs 0-31 */
+			"WIFI_RESET#",
+			"EN_WIFI_PWR",
+			"NA",
+			"NA",
+			"NA",
+			"FRAM_WP#",
+			"EN_CL_1",
+			"EN_CL_2",
+			"EN_CL_3",
+			"EN_CL_4",
+			"EN_ADC1_10V",
+			"EN_ADC2_10V",
+			"EN_ADC3_10V",
+			"EN_ADC4_10V",
+			"EN_SD_POWER",
+			"EN_USB_HOST_5V",
+			"EN_OFF_BD_5V",
+			"EN_CELL_MODEM_PWR",
+			"EN_NIMBEL_3.3V",
+			"EN_GPS_PWR#",
+			"EN_CAN_XVR#",
+			"EN_232_XVR",
+			"EN_LS_OUT_1",
+			"EN_LS_OUT_2",
+			"EN_LS_OUT_3",
+			"EN_LS_OUT_4",
+			"EN_LS_OUT_5",
+			"EN_LS_OUT_6",
+			"EN_LS_OUT_7",
+			"NA",
+			"NA",
+			"NA",
+			/* Inputs 32-63 */
+			"DIG_IN_1",
+			"DIG_IN_2",
+			"DIG_IN_3",
+			"DIG_IN_4",
+			"SD_BOOT_JMP#",
+			"DIO_1_IN",
+			"DIO_2_IN",
+			"DIO_3_IN",
+			"DIO_4_IN",
+			"DIO_5_IN",
+			"DIO_6_IN",
+			"DIO_7_IN",
+			"NA",
+			"NA",
+			"NA",
+			"N7",
+			"P7",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA",
+			"NA";
+	};
+};
+
+&snvs_poweroff {
+	status = "okay";
+};
+
+&snvs_rtc {
+	status = "disabled";
+};
+
+&wdog1 {
```

Comment:
Try and keep the `&peripheral {` sections in alphabetical order. I wasn't paying attention to this until now.